artiq/gui/entries.py

```diff
@@ -188,174 +220,193 @@ def update(value):
-        self.addWidget(randomize, 3, 1)
-
-        apply_properties(start)
-        start.setPrecision()
-        start.setRelativeStep()
-        apply_properties(stop)
-        stop.setPrecision()
-        stop.setRelativeStep()
+        disable_scroll_wheel(self.start)
+        self.addWidget(self.start, 0, 1)
+
+        self.npoints = QtWidgets.QSpinBox()
+        self.npoints.setMinimum(1)
+        self.npoints.setMaximum((1 << 31) - 1)
+        disable_scroll_wheel(self.npoints)
+        self.addWidget(self.npoints, 1, 1)
+
+        self.stop = ScientificSpinBox()
+        self.stop.setStyleSheet("QDoubleSpinBox {color:red}")
+        self.stop.setMinimumSize(110, 0)
+        disable_scroll_wheel(self.stop)
+        self.addWidget(self.stop, 2, 1)
+
+        self.randomize = QtWidgets.QCheckBox("Randomize")
+        self.addWidget(self.randomize, 3, 1)
+
+        apply_properties(self.start)
+        self.start.setPrecision()
+        self.start.setRelativeStep()
+        apply_properties(self.stop)
+        self.stop.setPrecision()
+        self.stop.setRelativeStep()
         apply_properties(scanner)
 
+        self.set_state(state)
         def update_start(value):
-            state["start"] = value*scale
+            state["start"] = value*self.scale
             scanner.setStart(value)
-            if start.value() != value:
-                start.setValue(value)
+            if self.start.value() != value:
+                self.start.setValue(value)
 
         def update_stop(value):
-            state["stop"] = value*scale
+            state["stop"] = value*self.scale
             scanner.setStop(value)
-            if stop.value() != value:
-                stop.setValue(value)
+            if self.stop.value() != value:
+                self.stop.setValue(value)
 
         def update_npoints(value):
             state["npoints"] = value
             scanner.setNum(value)
-            if npoints.value() != value:
-                npoints.setValue(value)
+            if self.npoints.value() != value:
+                self.npoints.setValue(value)
 
         def update_randomize(value):
             state["randomize"] = value
-            randomize.setChecked(value)
+            self.randomize.setChecked(value)
 
         scanner.startChanged.connect(update_start)
         scanner.numChanged.connect(update_npoints)
         scanner.stopChanged.connect(update_stop)
-        start.valueChanged.connect(update_start)
-        npoints.valueChanged.connect(update_npoints)
-        stop.valueChanged.connect(update_stop)
-        randomize.stateChanged.connect(update_randomize)
-        scanner.setStart(state["start"]/scale)
+        self.start.valueChanged.connect(update_start)
+        self.npoints.valueChanged.connect(update_npoints)
+        self.stop.valueChanged.connect(update_stop)
+        self.randomize.stateChanged.connect(update_randomize)
+        scanner.setStart(state["start"])
```

Comment:
Is this change and `setStop` correct? We divide by `scale` before.

Reply:
This is not correct. Should not be edited since the API is only concerned in changing the current value. Thanks for noticing. Already tested and updated the code.